docs/source/upgrade.rst

```diff
@@ -5,21 +5,51 @@ Upgrade Guide
 =============
 
 This is a list of changes introduced in each release that users should be aware of when migrating from older versions.
-Most changes are carefully designed not to break existing code; however changes that may possibly break them are highlighted with a box.
 
 CuPy v8
 =======
 
 Dropping Support of CUDA 8.0 and 9.1
 ------------------------------------
 
-Starting from CuPy v8, CUDA 8.0 and 9.1 are no longer supported.
-Use CUDA 9.0, 9.2, 10.0 or later.
+CUDA 8.0 and 9.1 are no longer supported.
+Use CUDA 9.0, 9.2, 10.0, or later.
+
+Dropping Support of NumPy 1.15 and SciPy 1.2
+--------------------------------------------
+
+NumPy 1.15 (or earlier) and SciPy 1.2 (or earlier) are no longer supported.
 
 Update of Docker Images
 -----------------------
 
-CuPy official Docker images (see :doc:`install` for details) are now updated to use CUDA 10.2 and Python 3.6.
+* CuPy official Docker images (see :doc:`install` for details) are now updated to use CUDA 10.2 and Python 3.6.
+* SciPy and Optuna are now pre-installed.
+
+CUB Support and Compiler Requirement
+------------------------------------
+
+CUB module is now built by default.
+You can enable the use of CUB by setting ``CUPY_ACCELERATORS="cub"`` (see :doc:`reference/environment` for details).
+
+Due to this change, g++-6 or later is required when building CuPy from the source.
+See :doc:`install` for details.
+
+The following environment variables are no longer effective:
+
+* ``CUB_DISABLED``: Use ``CUPY_ACCELERATORS`` as aforementioned.
+* ``CUB_PATH``: No longer required as CuPy uses CUB source tree bundled with CUDA (only when using CUDA 11.0 or later) or CuPy distribution.
```

Comment:
```suggestion
* ``CUB_PATH``: No longer required as CuPy uses either the CUB source bundled with CUDA (only when using CUDA 11.0 or later) or the one in the CuPy distribution.
```
I am fine with leaving the previous one, this is just style so I can't say which is better, pick up the one you like most 😁